lsp-mode.el

```diff
@@ -5515,40 +5515,34 @@ A reference is highlighted only if it is visible in a window."
                                       line
                                       character)))
 
-(defun lsp--get-document-symbols ()
+(defun lsp--get-document-symbols (&optional callback)
   "Get document symbols.
 
 If the buffer has not been modified since symbols were last
 retrieved, simply return the latest result.
 
-Else, if the request was initiated by Imenu updating its menu-bar
-entry, perform it asynchronously; i.e., give Imenu the latest
-result and then force a refresh when a new one is available.
+If CALLBACK is specified, call it when the document's symbols
+have been fetched. It implies
+`lsp--document-symbols-request-async'.
 
-Else (e.g., due to interactive use of `imenu' or `xref'),
-perform the request synchronously."
+Else (e.g., due to interactive use of `imenu' or `xref'), perform
+the request synchronously."
   (if (= (buffer-chars-modified-tick) lsp--document-symbols-tick)
       lsp--document-symbols
     (let ((method "textDocument/documentSymbol")
           (params `(:textDocument ,(lsp--text-document-identifier)))
           (tick (buffer-chars-modified-tick)))
-      (if (not lsp--document-symbols-request-async)
-          (prog1
-              (setq lsp--document-symbols (lsp-request method params))
-            (setq lsp--document-symbols-tick tick))
-        (lsp-request-async method params
-                           (lambda (document-symbols)
-                             (setq lsp--document-symbols document-symbols
-                                   lsp--document-symbols-tick tick)
-                             (lsp--imenu-refresh))
-                           :mode 'alive)
+      (if (or callback lsp--document-symbols-request-async)
+          (lsp-request-async method params
+                             (lambda (document-symbols)
+                               (setq lsp--document-symbols document-symbols
+                                     lsp--document-symbols-tick tick)
+                               (when callback (funcall callback)))
+                             :mode 'alive)
+        (setq lsp--document-symbols (lsp-request method params)
```

Comment:
this is regression - the previous code was returning symbols, the new one is returning tick.

Reply:
You are right - for some reason I wasn't using lsp-imenu when testing. I fixed that. `lsp-imenu` works for me.